src/tests/integration/test_generic_db_opeations.py

```diff
@@ -32,23 +33,46 @@ def mock_log_api(monkeypatch):
         ("ip_with_mariaDB", 3),
         ("ip_with_SQLite", 3),
         ("ip_with_duckDB", 3),
+        ("ip_with_MSSQL", 3),
     ],
 )
 def test_query_count(ip_with_dynamic_db, excepted, request):
-    ip_with_dynamic_db = request.getfixturevalue(ip_with_dynamic_db)
-    out_normal_query = ip_with_dynamic_db.run_line_magic(
-        "sql", "SELECT * FROM taxi LIMIT 3"
-    )
-    assert len(out_normal_query) == excepted
-
-    # Test query with --with & --save
-    ip_with_dynamic_db.run_cell(
-        "%sql --save taxi_subset --no-execute SELECT * FROM taxi LIMIT 3"
-    )
-    out_query_with_save_arg = ip_with_dynamic_db.run_cell(
-        "%sql --with taxi_subset SELECT * FROM taxi_subset"
-    )
-    assert len(out_query_with_save_arg.result) == excepted
+    if ip_with_dynamic_db == "ip_with_MSSQL":
+        # MSSQL doesn't have LIMIT
+        ip_with_dynamic_db = request.getfixturevalue(ip_with_dynamic_db)
+        out = ip_with_dynamic_db.run_line_magic(
+            "sql",
+            """
+            SELECT *
+            FROM taxi
+            ORDER BY 1
+            OFFSET 0 ROWS FETCH NEXT 3 ROWS ONLY
+            """,
+        )
+        # Test query with --with & --save
+        ip_with_dynamic_db.run_cell(
+            "%sql --save taxi_subset --no-execute SELECT * FROM taxi \
+            ORDER BY 1 OFFSET 0 ROWS FETCH NEXT 3 ROWS ONLY"
+        )
+        out_query_with_save_arg = ip_with_dynamic_db.run_cell(
+            "%sql --with taxi_subset SELECT * FROM taxi_subset"
+        )
+        assert len(out) == excepted
+        assert len(out_query_with_save_arg.result) == excepted
+    else:
+        ip_with_dynamic_db = request.getfixturevalue(ip_with_dynamic_db)
+        out = ip_with_dynamic_db.run_line_magic("sql", "SELECT * FROM taxi LIMIT 3")
+
+        # Test query with --with & --save
+        ip_with_dynamic_db.run_cell(
+            "%sql --save taxi_subset --no-execute SELECT * FROM taxi LIMIT 3"
+        )
+        out_query_with_save_arg = ip_with_dynamic_db.run_cell(
+            "%sql --with taxi_subset SELECT * FROM taxi_subset"
+        )
+
+        assert len(out) == excepted
+        assert len(out_query_with_save_arg.result) == excepted
```

Comment:
You can create a src/tests/integration/test_mssql.py to put exclusive test case

Reply:
Do you mean to remove the test cases :
```python
ip_with_dynamic_db.run_cell(
    "%sql --save taxi_subset --no-execute SELECT * FROM taxi \
    ORDER BY 1 OFFSET 0 ROWS FETCH NEXT 3 ROWS ONLY"
)
out_query_with_save_arg = ip_with_dynamic_db.run_cell(
  "%sql --with taxi_subset SELECT * FROM taxi_subset"
)
```
Because I think it is due to sqlglot. It doesn't support MSSQL

![image](https://user-images.githubusercontent.com/43580047/227027980-4d643fc2-ab29-48bc-b85a-ab12d87572fa.png)

Reply:
I already addressed in [110792b](https://github.com/ploomber/jupysql/pull/296/commits/110792b67d53783d281e6d6c0193269f685b23c7)

In `test_generic_db_operations.py`, we want to put the test cases together that can run universally (query, connection).

We will test some individual feature on specific database as a standalone file, e.g. `test_postgreSQL.py`, which has autocommit feature.

Same idea as mssql.

Reply:
That's the current way to keep our test cases clean